src/lib/tecs/TECS.cpp

```diff
@@ -607,3 +611,41 @@ void TECS::update_pitch_throttle(const matrix::Dcmf &rotMat, float pitch, float
 	}
 
 }
+
+void TECS::_update_speed_height_weights()
+{
+	// Calculate the weight applied to control of specific kinetic energy error
+	_SKE_weighting = constrain(_pitch_speed_weight, 0.0f, 2.0f);
+
+	if ((_underspeed_detected || _climbout_mode_active) && airspeed_sensor_enabled()) {
+		_SKE_weighting = 2.0f;
+
+	} else if (!airspeed_sensor_enabled()) {
+		_SKE_weighting = 0.0f;
+	}
+
+	// don't allow any weight to be larger than one, as it has the same effect as reducing the control
+	// loop time constant and therefore can lead to a destabilization of that control loop
+	_SPE_weighting = constrain(2.0f - _SKE_weighting, 0.f, 1.f);
+	_SKE_weighting = constrain(_SKE_weighting, 0.f, 1.f);
+}
+
+float TECS::SEB()
+{
+	return _SPE_estimate * _SPE_weighting - _SKE_estimate * _SKE_weighting;
+}
+
+float TECS::SEB_setpoint()
+{
+	return _SPE_setpoint * _SPE_weighting - _SKE_setpoint * _SKE_weighting;
+}
+
+float TECS::SEB_rate()
+{
+	return _SPE_rate * _SPE_weighting - _SKE_rate * _SKE_weighting;
+}
+
+float TECS::SEB_rate_setpoint()
+{
+	return _SPE_rate_setpoint * _SPE_weighting - _SKE_rate_setpoint * _SKE_weighting;
+}
```

Comment:
You could move all of these to their declaration

Reply:
Done